### PR TITLE
fix(overrides): avoid infinite recursion between `dask`, `distributed` and `dask-expr`

### PR DESCRIPTION
**Pull request by Atry**
[Contribution](README.md#contributing) checklist (recommended but not always applicable/required):

- [x] There's an _[automated test](tests/default.nix)_ for this change
- [x] Commit messages or code include _references to related issues or PRs_ (including third parties)
- [x] Commit messages are _[conventional](https://www.conventionalcommits.org/)_ - examples from [the log](https://github.com/nix-community/poetry2nix/commits/master) include "feat: add changelog files to fixup hook", "fix(contourpy): allow wheel usage", and "test: add sqlalchemy2 test"

### CODE REVIEW

File: tests/feast-wheel/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[tool.poetry]
+name = "feast-wheel"
+version = "0.1.0"
+description = ""
+authors = ["Star Dorminey <sterling.dorminey@gmail.com>"]
```
Comment (cpcloud) [inline tests/feast-wheel/pyproject.toml:5]: Please use a fake email address, like nearly every other test in this repository.

**Review by cpcloud — CHANGES_REQUESTED** [review]

File: tests/feast-wheel/pyproject.toml (hunk 1 above)
Comment (Atry) [reply tests/feast-wheel/pyproject.toml]: Fixed

**Review by cpcloud — DISMISSED** [review]

File: tests/feast-wheel/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[tool.poetry]
+name = "feast-wheel"
+version = "0.1.0"
+description = ""
+authors = ["Your Name <you@example.com>"]
+
+[tool.poetry.dependencies]
+python = "^3.10"
+feast = "^0.40.1"
```
Comment (cpcloud) [inline tests/feast-wheel/pyproject.toml:9]: Can you test **only** the `dask`/`dask-expr` dependency?

It's also fine to leave the `feast` test.

**Review by cpcloud — CHANGES_REQUESTED** [review]

File: tests/feast-wheel/pyproject.toml (hunk 2 above)
Comment (Atry) [reply tests/feast-wheel/pyproject.toml]: In fact, poetry2nix can build a project that depend on `dask` and `dask-expr` as long as no extra packages are specified for `dask`.

The infinite recursion happens only when another package, like `feast`, depends on `dask[dataframe]`, given that `dask-expr` depends on `dask`, and `dask[dataframe]` depends on `dask-expr`.

File: tests/overturemapsdownloader-wheel/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[tool.poetry]
+name = "overturemapsdownloader-wheel"
+version = "0.1.0"
+description = ""
+authors = ["Your Name <you@example.com>"]
+
+[tool.poetry.dependencies]
+python = ">=3.10,<4"
+overturemapsdownloader = "^0.1.9"
```
Comment (cpcloud) [inline tests/overturemapsdownloader-wheel/pyproject.toml:9]: This seems overkill versus a test that just depends on `dask = { ..., extras = ["dataframe"] }`.

I'll see if I can't fix that up and we can get this merged.

Comment (cpcloud) [reply tests/overturemapsdownloader-wheel/pyproject.toml]: Fixed.

**Review by cpcloud — APPROVED** [review]

**Pull request merged**